pkg/report/linux.go

```diff
@@ -701,8 +701,8 @@ func (ctx *linux) extractGuiltyFileRaw(title string, report []byte) string {
 		// we would need to ignore too many files and that would be fragile.
 		// So instead we try to extract guilty file starting from the known
 		// interrupt entry point first.
-		for _, interruptEnd := range []string{" apic_timer_interrupt+0x", "Exception stack"} {
-			if pos := bytes.Index(report, []byte(interruptEnd)); pos != -1 {
+		for _, interruptEnd := range []string{"apic_timer_interrupt+0x", "Exception stack"} {
+			if pos := bytes.LastIndex(report, []byte(interruptEnd)); pos != -1 {
```

Comment:
This makes sense for the current stack, but can't there be multiple stacks from different CPUs/tasks and then we will take the very last stack at the bottom.
If we skip apic_timer_interrupt frames, then finding the last is not necessary.

Reply:
The problem is that, while determining the guilty files, we skip files rather than patterns. Ideally we should be using the same code and rules, but it's a much bigger refactoring that seems reasonable at the moment.

I've added the x86 apic code to the skipped guilty files and it began to work without `LastIndex`. I tried to find some non-x86 rcu stalls to see what files need to be skipped in that case, but didn't manage to find anything like that on syzbot.